### PR TITLE
Harden unit tests against spaces in file pathes

### PR DESCRIPTION
**Pull request by AustinWise**
Some unit tests fail if there are spaces in `%USERPROFILE%`. These changes fix some (but not all, as there are a lot) unit tests to run without error.

### CODE REVIEW

File: src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs [hunk 1]
```diff
@@ -186,7 +186,21 @@ private static (T Result, string Output) UseTextWriter<T>(Encoding encoding, Fun
             Encoding redirectEncoding = null,
             bool shouldRunOnServer = true)
         {
-            var arguments = new List<string>(argumentsSingle.Split(' '));
+            var arguments = argumentsSingle.Split(' ');
+            return RunCommandLineCompiler(language, arguments, currentDirectory, filesInDirectory,
+                additionalEnvironmentVars, redirectEncoding, shouldRunOnServer);
+        }
```
Comment (jmarolf) [inline src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs]: Why this change?

**Comment by jaredpar** [comment]
Spaces in the path are not supported for this repository. This issue tracks the state

https://github.com/dotnet/roslyn/issues/20929

In general we don't want to take changes to fix this unless the infrastructure is changed at the  same time such that there are spaces in the path during CI. If it's not tested, it will be broken again almost immediately.

**Comment by AustinWise** [comment]
I was kinda guessing that, but could not find it the docs. I’ll close this, as it is probably easier to change users names.

**Comment by jasonmalinowski** [comment]
@jaredpar Is it worth keeping this PR open once we sort out the validation questions of it?

@AustinWise We're having an internal conversation of whether we can update our CI to put spaces in the paths, at which point it means we can actually validate this scenario.

**Comment by AustinWise** [comment]
@jasonmalinowski For what it's worth, if you create a local account (non-domain, non-Microsoft login), Windows setup prompts you "Who is going to use this?". If you type your full name in there, you end up with the space in `%USERPROFILE%`. So it may be a common-ish occurrence to have spaces.
